src/main/java/com/crud/cadastrousuario/domain/service/HistoricService.java

```diff
@@ -0,0 +1,86 @@
+package com.crud.cadastrousuario.domain.service;
+
+import com.crud.cadastrousuario.domain.dto.HistoricDTO;
+import com.crud.cadastrousuario.domain.dto.LoanOfBooksDTO;
+import com.crud.cadastrousuario.domain.dto.UserDTO;
+import com.crud.cadastrousuario.domain.exception.BadRequestException;
+import com.crud.cadastrousuario.domain.model.Historic;
+import com.crud.cadastrousuario.domain.model.LoanOfBooks;
+import com.crud.cadastrousuario.domain.model.User;
+import com.crud.cadastrousuario.domain.repository.HistoricRepository;
+import com.crud.cadastrousuario.domain.repository.HistoricSpec;
+import com.crud.cadastrousuario.domain.repository.LoanOfBooksRepository;
+import com.crud.cadastrousuario.domain.repository.UserRepositorySpec;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Log4j2
+@Service
+public class HistoricService {
+
+
+    @Autowired
+    private LoanOfBooksRepository loanOfBooksRepository;
+
+    @Autowired
+    private HistoricRepository historicRepository;
+
+    private static final double AMOUNT_OF_THE_FINE_WITHOUT_DELAY = 0.0;
+
+    private static final double FINE_RATE_DAYS = 2.15;
```

Comment:
o que é essa regra de negocio?

Reply:
Essa regra é sobra a multa de alguel de livro. Dessa forma, quando for entregue de forma atrasada paga uma taxa de 2.15 por dia atrasado e se não tiver atrasado paga nada